### PR TITLE
Parse 17-point annual data for MotiDataController.

### LINKED ISSUE #50: MOTI portal does not properly display annual timeseries data
@jameshiebert says:

> This is basically a problem with the DataController asking for and receiving the wrong type of data (or at least not checking what type of data it's asking for).
> 
> It's getting a 17-point seasonal climatology (i.e. 12 months + 4 seasons + 1 annual), but it's trying to display it as timeseries plot (which it shouldn't).
> 
> I think the solution to this is having the DataController check the metadata more carefully.

### PR DESCRIPTION
**Pull request by corviday**
The parse function used by MotiDataController, `chart.timeSeriesToC3()`, and the parse function used by DataController (`util.parseTimeSeriesForC3()`) were so similar (and would become much moreso with the addition of seventeen-point-chronology to timeSeriesToC3) I just got rid of `chart.js` entirely and added an option to `parseTimeSeriesForC3()` to generate a MoTI-style chart without seasonal or yearly means.

Fixes #50 and #48.

### CODE REVIEW

File: src/core/__tests__/util-test.js [hunk 1]
```diff
@@ -183,14 +183,19 @@ const outputC3TimeSeriesTooltipInfo = {
 };
 
 describe('parseTimeSeriesForC3', function () {
-  it('Correctly parses a JSON object with time series data from one model for plotting with C3', function () {
-    var parseTimeSeriesForC3 = require('../util').parseTimeSeriesForC3;
+  it('Correctly parses a JSON object with time series data from one model for plotting with C3',
```
Comment (jameshiebert) [inline src/core/__tests__/util-test.js]: So, I realize that this test was named as such before you worked on this code. However, one thing that I find very non-helpful is when tests vaguely refer to their subjects producing "correct" behaviour, without really defining what correct means. No action required here, but keep that in mind as you move forward with this code base.

**Comment by rod-glover** [comment]
I agree on all points. The word 'correctly' is code smell in tests!
 
When the time comes, I suggest this function be  broken up into component functions that construct various parts, such as the axis info, and those component functions  tested separately and explicitly. On a shallow pass, it looks to me like there should be several such functions, each of which should have some reasonably compact specification (test set). 

It's possible that client code may be better to assemble them locally rather than to rely on an uber function to do that based on flags etc.

**Comment by corviday** [comment]
Does it make sense for me to take a crack at refactoring `parseTimeSeriesForC3()` now, since I currently understand how it works?

**Comment by corviday** [comment]
[Docker test for this branch](http://142.104.230.42:8302/moti). 
Known issue: stats table doesn't load data, due to #53. 

I don't think we were 100% on whether we wanted to continue the status quo (no seasonal or yearly averages on the MOTI portal datagraph), or have the MOTI portal display the same seasonal and annual data as [the other portal's annual series data](http://142.104.230.42:8302/climo).

The current state of the branch preserves the status quo, but it would be incredibly easy to have all the Annual Series Datagraphs display data the same way if, after looking at the test version, we decided we prefer that.

**Comment by jameshiebert** [comment]
Hmmmm, something is wrong with the docker test that you have above. Precip shows up as being highest during the spring (not winter) and temperature shows up as being highest during the winter (not summer). Can you double check your time processing?

**Comment by corviday** [comment]
Hm. That is rather odd.
As far as I can tell, the frontend is accurately displaying what it gets from the backend. Here's the graph from the same model, different area, showing precipitation highest in June:

screenshot from 2017-06-09 09-18-05

[Here's the relevant API call to the backend](http://142.104.230.42:8300/api/timeseries?id_=pr_Amon_CSIRO-Mk3-6-0_rcp85_r1i1p1_20100101-20391231&variable=pr&area=):

```json
{
"id": "pr_Amon_CSIRO-Mk3-6-0_rcp85_r1i1p1_20100101-20391231", 
"data": 
  { 
  "2025-01-16T00:00:00Z": 2.2985157436794705, 
  "2025-02-15T00:00:00Z": 2.3018962012396917, 
  "2025-03-16T00:00:00Z": 2.3259194691975913, 
  "2025-04-16T00:00:00Z": 2.381052017211914, 
  "2025-05-16T00:00:00Z": 2.4425648583306208, 
  "2025-06-16T00:00:00Z": 2.499209509955512, 
  "2025-07-16T00:00:00Z": 2.4914461771647134, 
  "2025-08-16T00:00:00Z": 2.4258770412868924, 
  "2025-09-16T00:00:00Z": 2.3558497958713107, 
  "2025-10-16T00:00:00Z": 2.317519293891059, 
  "2025-11-16T00:00:00Z": 2.3151658376057944, 
  "2025-12-16T00:00:00Z": 2.3070203993055554, 
  "2025-01-15T00:00:00Z": 2.302496804131402, 
  "2025-04-17T00:00:00Z": 2.3832018110487194, 
  "2025-07-17T00:00:00Z": 2.471883773803711, 
  "2025-10-17T00:00:00Z": 2.3293800354003906, 
  "2025-07-02T00:00:00Z": 2.372236039903429
  }, 
"units": "mm"
}
```

June precipitation is also highest there. So maybe I am using a bad dataset in the backend docker?

**Comment by jameshiebert** [comment]
> As far as I can tell, the frontend is accurately displaying what it gets from the backend

Ah, yep you're right.

> So maybe I am using a bad dataset in the backend docker?

You *should* be able to trust this, but I'll look into it and make sure everything's kosher on the backend. I guess feel free to ignore this for now and take the backend as gospel.

> Does it make sense for me to take a crack at refactoring parseTimeSeriesForC3() now, since I currently understand how it works?

Sure, I think that makes sense, if you're in the right head space to do it now as part of this ticket. Poke me when you feel like it's in good shape and I'll re-review.

**Comment by corviday** [comment]
I'd like us to merge this one now, if it all looks good. 

I was planning to refactor `parseDataForC3()`, the function that generates graphs from the API JSON results, as part of this branch. But we're getting rid of seventeen-point chronology data files entirely, so any changes I make to `parseDataForC3()` at this point are going to be immediately outdated. I'd like to merge this now so I can build on it for the next round of data changes.

**Comment by jameshiebert** [comment]
Sounds good. I'll go ahead a merge.

**Pull request merged**